SG14_test/slot_map_test.cpp

```diff
@@ -446,6 +446,52 @@ void BoundsCheckingTest()
     assert(cit == sm.end());
 }
 
+template<class SM>
+void GenerationTest()
+{
+    using T = typename SM::mapped_type;
+
+    {
+        SM sm;
+
+        for (int i = 0; i < 10; ++i) {
+            auto k = sm.insert(Monad<T>::from_value(i));
```

Comment:
The problem with this test for a LIFO would be that this line (line 458) assumes that "default-construct the container; push_back one element onto it" leaves the container with capacity equal to 1. If the container's capacity started at >=2, or if it preemptively jumped up to >=2 as soon as you put one thing in it, then we'd have more slots at our disposal here than just the one. And so this test is relying on _either_ (what I just described, not happening) _or_ (FIFO behavior for indices).

I believe the test starting on line 472 avoids this issue because it carefully uses up all the already-reserved slots after the first (required) insertion.

Reply:
I see, that's a good observation. I'll tweak your test a little bit.